### DISCUSSION TITLE
Is it necessary to check message event origin for extension to webview comunication.

### DISCUSSION THREAD

**Discussion by rafalou38** [microsoft/vscode-discussions#1061] [Extension Development QnA] [answered]
I am the maintainer of a vscode extension that previews code in the webview.
To allow communication between the extension and the webview I introduced this code in the webview context:
```ts
window.addEventListener("message", (event) => {
  if (!event.origin.includes("vscode-webview"))
    return console.warn("wrong origin: " + event.origin);
  // ...
});
```
Extension context:
```ts
this._panel.webview.postMessage({
      type: "codeUpdate",
      value: result,
    });
```

Some users stumble across an issue when using the extension online where event.origin does not start with "vscode-webview:" but "https:".
So I am wondering, is this check really necessary ? And if it is, how can I make it work in every situation.

**Comment by theReynald** [comment, marked as answer]
Hi @rafalou38 

To make your extension work in every situation while securely, you can adjust your origin check to allow for different origins. One way to achieve this is by checking if the origin starts with vscode-webview: OR if it starts with https:. Try...

```JavaScript
window.addEventListener("message", (event) => {
  if (
    !event.origin.startsWith("vscode-webview:") &&
    !event.origin.startsWith("https:")
  ) {
    return console.warn("Wrong origin: " + event.origin);
  }
  // Handle messages from trusted origins
});
```

This modification allows messages from both vscode-webview: and https: origins, ensuring that your extension will work both locally within VS Code and when hosted online. Let me know if this helps.

**Reply by rafalou38** [reply]
Thanks, that should work. I was just unsure if allowing any https was safe.

**Reply by isidorn** [reply]
@mjbvz might have advice here

**Reply by mjbvz** [reply]
I believe you should check that `e.origin === window.origin` instead of hardcoding any strings
